### PR TITLE
LPD-31855 Avoid deadlock when editing picklist object field from non-published object definitions

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPD-31855

### CODE REVIEW

**Review by carolmariaabb — APPROVED** [review]
BE changes -> LGTM 🎉

**Comment by marcelabc** [comment]
Hey @rebsilva 
Could you review this PR, please? :smile:

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:playwright

**Comment by rebsilva** [comment]
just started reviewing! :cherry_blossom:

File: modules/test/playwright/tests/object-web/objectFields.spec.ts [hunk 1]
```diff
@@ -327,6 +327,75 @@ test.describe('Manage object fields through Model Builder', () => {
...
+		);
+
+		await modelBuilderPage.clickShowAllFieldsButton(
+			draftObjectDefinition.label['en_US']
+		);
+
+		await page.getByText(picklistFieldName).click();
+
+		picklistFieldName = 'picklistField' + getRandomInt();
+		await page
```
Comment (rebsilva) [inline modules/test/playwright/tests/object-web/objectFields.spec.ts:385]: can you put a space between these lines please?

File: modules/test/playwright/tests/object-web/objectFields.spec.ts [hunk 2]
```diff
@@ -327,6 +327,75 @@ test.describe('Manage object fields through Model Builder', () => {
...
+				status: {code: 2},
+			});
+
+		const listTypeDefinition =
+			await apiHelpers.listTypeAdmin.postRandomListTypeDefinition();
+
+		listTypeDefinitionIds.push(listTypeDefinition.id);
+
+		let picklistFieldName = 'picklistField' + getRandomInt();
+		await apiHelpers.objectAdmin.postObjectFieldByExternalReferenceCode(
```
Comment (rebsilva) [inline modules/test/playwright/tests/object-web/objectFields.spec.ts:350]: can you put a space between these lines please?

**Review by rebsilva — CHANGES_REQUESTED** [review]
hi @p-albuquerque! great work on the test, just asked for some little adjustments :cherry_blossom:

**Comment by p-albuquerque** [comment]
ci:test:sf

**Review by rebsilva — APPROVED** [review]
LGTM! :tada:

**Comment by carolmariaabb** [comment]
Playwright test is passing
image

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:object

**Comment by p-albuquerque** [comment]
ci:report:44686397

**Comment by p-albuquerque** [comment]
There are three poshi tests with failures related to Picklist that I intend to carefully analyze:
LocalFile.ObjectFieldDefaultValue#CanOnlyAffectNewEntriesByUpdatingConfiguration
LocalFile.ObjectFieldDefaultValue#CanSetPicklistViaExpressionBuilder
LocalFile.ObjectFieldDefaultValue#CanSetPicklistViaInputAsValue

**Comment by p-albuquerque** [comment]
The failures below is related to this fix:
LocalFile.ObjectFieldDefaultValue#CanOnlyAffectNewEntriesByUpdatingConfiguration

I'm working to overcome it...

**Comment by p-albuquerque** [comment]
ci:report:45565103

**Comment by p-albuquerque** [comment]
ci:forward:force

**Comment by p-albuquerque** [comment]
ci:forward

**Pull request closed without merging**